### PR TITLE
[ready-for-acceptance, work in progress] Postcode search endpoint

### PR DESCRIPTION
**Pull request by Pheonnexx**

### CODE REVIEW

File: service/server.py [hunk 1]
```diff
@@ -26,6 +76,17 @@ def get_title(title_ref):
         #Title not found
         abort(404)
 
+@app.route('/title_search_postcode/<postcode>', methods=['GET'])
+def get_properties(postcode):
+    postcode.replace ("_", " ")
```
Comment (camilleldn) [inline service/server.py]: Are we sure this is the sanitising we need to do here? :) I am expecting a lot more sanitising going on potentially in the digital-register-frontend, then passing a postcode that we know is in an acceptable format here?

**Comment by camilleldn** [comment]
LGTM so far, happy to walk you through an interactive rebase if you want to modify your commit messages :) let's see what happens when we plug it into an actual elasticsearch thing :)

File: service/server.py [hunk 2]
```diff
@@ -1,17 +1,57 @@
...
+    client = Elasticsearch([ELASTIC_SEARCH_ENDPOINT])
+    search = Search(using=client, index='landregistry', doc_type='property_by_postcode')
+    query = search.filter('term', postcode=postcode)
+    return query.execute().hits
+
+def format_address_into_single_string(address_record):
+    address = [address_record.buildingNumber, address_record.thoroughfareName, address_record.postTown,
+        address_record.postCode]
+    formatted_address = ", ".join(address)
+    return formatted_address
```
Comment (Pheonnexx) [inline service/server.py]: I think this function can be removed as it is being no longer used?

File: service/server.py [hunk 3]
```diff
@@ -26,6 +66,17 @@ def get_title(title_ref):
         #Title not found
         abort(404)
 
+@app.route('/title_search_postcode/<postcode>', methods=['GET'])
+def get_properties(postcode):
+    postcode.replace ("_", " ")
+    address_records = get_property_address(postcode)
+    if address_records:
+        result = format_address_records(address_records)
+        print(result)
```
Comment (aronjasper) [inline service/server.py]: This 'print' should probably be removed.

File: service/server.py (hunk 1 above)
Comment (camilleldn) [inline service/server.py]: Need to talk Ian about this one :(

**Pull request closed without merging**